lib/storage/storage.go

```diff
@@ -491,8 +500,19 @@ func (s *Storage) DeleteStaleSnapshots(maxAge time.Duration) error {
 	return nil
 }
 
-func (s *Storage) idb() *indexDB {
-	return s.idbCurr.Load()
+// getIndexDBs increments refcount for the current and next indexDB and returns
+// them along with a cleanup function that decrements their refcounts.
+func (s *Storage) getIndexDBs() (*indexDB, *indexDB, func()) {
```

Comment:
A bit weird that this func is used 32 times, and only once the `nextIdx` is actually needed. Did you consider splitting it into two funcs?